kafka/admin/client.py

```diff
@@ -605,42 +621,49 @@ def describe_consumer_groups(self, group_ids, group_coordinator_id=None):
             partition assignments.
         """
         group_descriptions = []
+        futures = []
         version = self._matching_api_version(DescribeGroupsRequest)
         for group_id in group_ids:
-            if group_coordinator_id is not None:
-                this_groups_coordinator_id = group_coordinator_id
-            else:
-                this_groups_coordinator_id = self._find_group_coordinator_id(group_id)
             if version <= 1:
+                if group_coordinator_id is not None:
+                    this_groups_coordinator_id = group_coordinator_id
+                else:
+                    this_groups_coordinator_id = self._find_group_coordinator_id(group_id)
```

Comment:
Is there a reason you moved this under the version check? This seems generic code that's currently needed regardless of the version.

Side note: `_find_group_coordinator_id()` is a blocking call so this could be further optimized into firing off a bunch of futures for all the groups and then handling the responses but that would require breaking `_find_group_coordinator_id()` into two methods to generate request/parse the response... that optimization is probably best handled in different PR. Especially because that wouldn't even be used when fetching offsets for all consumer groups since the group coordinator is explicitly passed in already... example: https://github.com/DataDog/integrations-core/pull/2730/files#diff-deed983f73a04522ac621d7f6d0c8403R244

Reply:
Figured it could be a minor optimization.  We don't actually use this_groups_coordinator_id unless version <= 1 evaluates true, so why initialize it or call _find_group_coordinator_id() if we don't have to?

I'll peek at _find_group_coordinator_id() when I have a spare minute and see if I can optimize it.  Agreed, something for another PR, but shouldn't be too bad.